addons/sale/report/ir_actions_report_templates.xml

```diff
@@ -227,7 +227,10 @@
         <t t-set="is_pro_forma" t-value="True"/>
         <t t-set="docs" t-value="docs.with_context(proforma=True)"/>
         <t t-foreach="docs" t-as="doc">
-            <t t-call="sale.report_saleorder_document" t-lang="doc.partner_id.lang"/>
+            <t t-if="doc._get_name_proforma_report() == 'sale.report_saleorder_document'"
+               t-call="sale.report_saleorder_document"
+               t-lang="doc.partner_id.lang"
+            />
```

Comment:
I just remembered https://github.com/odoo/odoo/commit/22c47fb003c7cfaccb732de5b8ea7277b9dade54
and honestly, I believe it's a cleaner approach if you need to override stuff in localization modules (with some xpaths instead of adding a new method here :eyes:

Reply:
@Feyensv That's not so easy to override stuff in localization modules...
1) we have to deal with the stable policy
2) it's not clean/difficult to debug when there is too much logic in the template

For example:
```py
    def _get_name_invoice_report(self):
        # EXTENDS account
        self.ensure_one()
        if self.l10n_mx_edi_cfdi_state == 'sent' and self.l10n_mx_edi_cfdi_attachment_id:
            return 'l10n_mx_edi.report_invoice_document'
        return super()._get_name_invoice_report()
```

3) it's cleaner than doing this in all localizations:

```xml
    <template id="report_saleorder_pro_forma" inherit_id="sale.report_saleorder_pro_forma">
        <xpath expr='//t[@t-call="sale.report_saleorder_document"]' position="attributes">
            <attribute name="t-if" add="doc.company_id.country_code != 'MX'" separator=" and "/>
        </xpath>
        <xpath expr='//t[@t-call="sale.report_saleorder_document"]' position="after">
            <t t-elif="doc.company_id.country_code == 'MX'"
               t-call="l10n_mx_edi_sale.report_saleorder_document_proforma" t-lang="doc.partner_id.lang"/>
        </xpath>
    </template>
```
... here it's only about the country_code but it won't be so simple everytime ;)

Having python methods returning all the values needed to render a template is a better approach.

Reply:
@Feyensv Less clean, less lisible. Btw dynamic t-call is not a bad practice. We can’t use them because they are not supported by studio. It’s one of the numerous things that is not well managed by this module… I hope this will be fixed one day. Let’s do it in your way this time.